File: CRT CODING PRACTISE/08 May/prg1.py
n = int(input())

arr = list(map(int, input().split()))

k = int(input())

max_len = 0

for i in range(n):

    total = 0

    for j in range(i, n):

        total += arr[j]

        if total == k:
            max_len = max(max_len, j - i + 1)

print(max_len)

File: CRT CODING PRACTISE/08 May/prg1.c

int main() {

    int n;
   // scanf("%d", &n);

    int arr[7];//n

    for(int i = 0; i < n; i++) {
        scanf("%d", &arr[i]);
    }

    int k;
    scanf("%d", &k);

    int maxLen = 0;

    for(int i = 0; i < n; i++) {

        int sum = 0;

        for(int j = i; j < n; j++) {

            sum += arr[j];

            if(sum == k) {

                int length = j - i + 1;

                if(length > maxLen) {
                    maxLen = length;
                }
            }
        }
    }

    printf("%d", maxLen);

    return 0;
}